Automattic/wp-calypso
# VideoPress Onboarding: add support for an optional "more" button on Intro step

#### Proposed Changes

* Add support for an **_optional_** button that pops a modal on the Intro page for all onboarding flows
* adds VideoPress custom modal content

![SCR-20230206-mj9](https://user-images.githubusercontent.com/4081020/217089418-8c29fe5d-8946-44c6-b9e3-3f813a40b068.jpeg)

| desktop | mobile |
| --- | --- |
| <img width="1592" alt="SCR-20230206-mfe" src="https://user-images.githubusercontent.com/4081020/217089196-1a5cba02-2f28-4774-8754-b8780b464658.png"> | <img width="549" alt="SCR-20230206-mf6" src="https://user-images.githubusercontent.com/4081020/217089191-3911beec-7cf1-4b33-b577-c7b092425b65.png"> |


NOTE: The `z-index` on the logo bar (`signup-header`) component is overridden to 0 on the `intro` page only to prevent it from overlapping the new modal. This change appears to work OK across all flows that use the `intro` flow (link-in-bio and newsletter). It isn't clear why this was initially set to `1` but it is `-1` in the original component, so presumably it was to fix an issue on one of the other steps.

cc @Automattic/vertex as it appears your team has modified these files recently.

#### Testing Instructions



* pull PR
* `yarn && yarn start`
* Visit `/setup/videopress`
* You should see the "Learn More" button
* Clicking it should open the modal
* Clicking `X` should close the modal
* In the modal, the links should all go somewhere
   *  Videomaker -> Videomaker theme page (jetpack because the wpcom one forces you to choose a site first)
   * Videopress -> videopress.com
   * Get Started -> next step in the flow??
   * "or learn more about videopress" -> videopress.com
* Test the "generic-ness" of the modal  by adding the following code snippet to the Newsletter flow in `intro/index.tsx` around line 47
```
modal: {
    buttonText: __( 'HI' ),
    content: null,
},
```
* then view the flow at `/setup/newsletter` -> the button should work as expected, but with no content
* Test the intro pages of other flows to ensure that the logo appears correctly
    * `/setup/link-in-bio`
    * `/setup/newsletter`

#### Pre-merge Checklist





Related to #
